presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -101,6 +104,10 @@
     protected final String connectionUrl;
     protected final Properties connectionProperties;
     protected final String identifierQuote;
+    private final boolean mapTableNames;
+
+    private Map<String, Map<String, String>> schemaTableMapping = new HashMap<>();
```

Comment:
I do not like Map of Map things, can you please extract some class instead? That class could have different implementation for `mapTableNames` equal to `true` and `false`. You could have a lock there as well.